x-pack/plugins/ml/public/embeddables/anomaly_explorer/anomaly_explorer_initializer.tsx

```diff
@@ -0,0 +1,125 @@
+  onCreate: (props: { panelTitle: string; maxSeriesToPlot?: number }) => void;
+  onCancel: () => void;
+}
+
+export const AnomalyExplorerInitializer: FC<AnomalyExplorerInitializerProps> = ({
+  defaultTitle,
+  initialInput,
+  onCreate,
+  onCancel,
+}) => {
+  const [panelTitle, setPanelTitle] = useState(defaultTitle);
+  const [maxSeriesToPlot, setMaxSeriesToPlot] = useState(
+    initialInput?.maxSeriesToPlot ?? DEFAULT_MAX_SERIES_TO_PLOT
+  );
+
+  const isPanelTitleValid = panelTitle.length > 0;
+
+  const isFormValid = isPanelTitleValid && maxSeriesToPlot > 0;
+  return (
+    <EuiModal initialFocus="[name=panelTitle]" onClose={onCancel}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.ml.anomalyExplorerChartsEmbeddable.setupModal.title"
+            defaultMessage="Anomaly explorer charts configuration"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiForm>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.anomalyExplorerChartsEmbeddable.panelTitleLabel"
+                defaultMessage="Panel title"
+              />
+            }
+            isInvalid={!isPanelTitleValid}
+          >
+            <EuiFieldText
+              data-test-subj={'panelTitleInput'}
+              id="panelTitle"
+              name="panelTitle"
+              value={panelTitle}
+              onChange={(e) => setPanelTitle(e.target.value)}
+              isInvalid={!isPanelTitleValid}
+            />
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.anomalyExplorerChartsEmbeddable.maxSeriesToPlotLabel"
+                defaultMessage="Max series to plot"
+              />
+            }
+          >
+            <EuiFieldNumber
+              id="selectMaxSeriesToPlot"
+              name="selectMaxSeriesToPlot"
+              value={maxSeriesToPlot}
+              onChange={(e) => setMaxSeriesToPlot(parseInt(e.target.value, 10))}
+              min={0}
+              max={MAX_SERIES_ALLOWED}
+            />
+          </EuiFormRow>
+        </EuiForm>
+      </EuiModalBody>
+
+      <EuiModalFooter>
+        <EuiButtonEmpty onClick={onCancel} data-test-subj={'cancelButton'}>
+          <FormattedMessage
+            id="xpack.ml.anomalyExplorerChartsEmbeddable.setupModal.cancelButtonLabel"
+            defaultMessage="Cancel"
+          />
+        </EuiButtonEmpty>
+
+        <EuiButton
+          data-test-subj={'confirmButton'}
+          isDisabled={!isFormValid}
+          onClick={onCreate.bind(null, {
+            panelTitle,
+            maxSeriesToPlot,
+          })}
+          fill
+        >
+          <FormattedMessage
+            id="xpack.ml.anomalyExplorerChartsEmbeddable.setupModal.confirmButtonLabel"
+            defaultMessage="Confirm"
```

Comment:
We should try to [label buttons with their action](https://eui.elastic.co/#/guidelines/writing) - what are we confirming?

Reply:
Updated here [`cb43b45` (#94396)](https://github.com/elastic/kibana/pull/94396/commits/cb43b4560e8dc54fe35aa2a9efffdd11b5c559e3)